lib/phobos_db_checkpoint/failure.rb

```diff
@@ -0,0 +1,51 @@
+module PhobosDBCheckpoint
+  class Failure < ActiveRecord::Base
+    def self.record(event_payload:, event_metadata:, exception: nil)
+      return if exists?(event_metadata[:checksum])
+
+      create do |record|
+        record.payload         = event_payload
+        record.metadata        = event_metadata
+        record.error_class     = exception&.class&.name
+        record.error_message   = exception&.message
+        record.error_backtrace = exception&.backtrace
+      end
+    end
+
+    def self.exists?(checksum)
+      where("metadata->>'checksum' = ?", checksum).exists?
+    end
+
+    def payload
+      attributes['payload'].deep_symbolize_keys
+    end
+
+    def metadata
+      attributes['metadata'].deep_symbolize_keys
+    end
+
+    # Can we delete the failure already at this stage?
```

Comment:
Thoughts about this are welcome. Like I write, I am thinking retry! should delete the record. It will be  created again if it winds up failing enough times again.